datamonster_api/lib/datamonster.py

```diff
@@ -247,3 +256,219 @@ def _avro_to_df(self, avro_buffer):
+                                       'country': 'US',
+                                       'section_pk': 707}}]
+
+        :raises: DataMonsterError if `filters` is not a dict or is not JSON-serializable.
+            Re-raises `DataMonsterError` if self.client.get() raises that.
+        """
+        self._check_param(datasource=datasource)
+
+        params = {}
+        if filters:
+            params['filters'] = self.to_json_checked(filters)
+        if self.SPLITS_PAGESIZE:
+            params['pagesize'] = self.SPLITS_PAGESIZE
+
+        url = self._get_dimensions_path(uuid=datasource.id)
+        if params:
+            url = ''.join([url, '?', six.moves.urllib.parse.urlencode(params)])
+
+        # Let any DataMonsterError from self.client.get() happen -- we don't occlude them
+        return DimensionSet(url, self, _pk2ticker=_pk2ticker)
+
+    def _convert_section_pks_to_tickers(self, dimension, pk2ticker_memos):
+        """
+        :param dimension: a dimension dict, with a key 'split_combination'.
+        :param pk2ticker_memos: (dict) maps section_pk => ticker
+            section_pk's already in this dict will use the tickers already looked up and saved;
+            new section_pk's will have their tickers saved here
+
+        :return: the dict, mutated:
+            if 'section_pk' in result['split_combination'], its value
+                result['split_combination']['section_pk"]
+            is a list of section_pk's (though we acommodate a single pk or None);
+            we replace each pk with
+                self._pk_to_ticker(pk)          if ticker is not None,
+                with str(pk) + '-NO_TICKER'     if ticker is None
+        """
+        combo = dimension['split_combination']
+        if 'section_pk' in combo:
+            value = combo.pop('section_pk')
+            if value is not None:
+                combo['ticker'] = (
+                    self._pk_to_ticker(value, pk2ticker_memos)
+                    if isinstance(value, int) else          # isinstance(value, list) -- List[int] in fact
+                    six.moves.map(lambda pk: self._pk_to_ticker(pk, pk2ticker_memos), value)
+                )
+        return dimension
+
+    def _pk_to_ticker(self, pk, pk2ticker_memos):
+        """
+        :param pk: int -- a section_pk
+        :param pk2ticker_memos: (dict) maps section_pk => ticker
+            section_pk's already in this dict will use the tickers already looked up and saved;
+            newly-encountered section_pk's will have their tickers saved here
+
+        :return: str --
+                dm.get_company_from_pk(pk).ticker   if that is not None,
+                str(pk) + '-NO_TICKER'              otherwise (actual ticker is None or empty)
+        """
+        if pk not in pk2ticker_memos:
+            ticker = self.get_company_by_pk(pk).ticker
+            if not ticker:
+                ticker = str(pk) + '-NO_TICKER'
```

Comment:
i don't understand the purpose of this string concat

Reply:
Maybe just use the company name instead like we do on the normal site?

Reply:
It's *something*, in the event the `pk` has no ticker (there _are_ tickerless `section_pk`s). Simply using str(pk) might be confusing; this at least is a string with no spaces, which doesn't look like a ticker.
No harm in doing what DM does (using the company name). Done.